app/charts/chart-data-wrapper.spec.tsx

```diff
@@ -56,22 +56,26 @@ jest.mock("@/graphql/hooks", () => ({
 
 jest.mock("@/components/hint", () => ({
   Loading: jest.fn(() => null),
+  NoDataHint: jest.fn(() => null),
 }));
 
 describe("ChartDataWrapper", () => {
-  it("should not render the component if prop is still being loaded", () => {
+  it("should render the LoadingOverlay if prop is still being loaded", () => {
     const Chart = jest.fn();
+    const LoadingOverlay = jest.fn();
     render(
       <LoadingStateProvider>
         <ChartDataWrapper
           chartConfig={{ cubes: [] } as any as ChartConfig}
+          LoadingOverlayComponent={LoadingOverlay}
           Component={Chart}
           dataSource={{ type: "sparql", url: "url" }}
           observationQueryFilters={[]}
           fetching
         />
       </LoadingStateProvider>
     );
-    expect(Chart).not.toHaveBeenCalled();
+    expect(Chart).toHaveBeenCalled();
+    expect(LoadingOverlay).toHaveBeenCalled();
```

Comment:
In testing-library style (testing what the user sees), I think we would have text.
This way for example if the LoadingOverlay ends up being in a display: none container, the test would fail.

```
const Chart = () => <div>My chart</div>
const LoadingOverlay = () => <div>Loading overlay</div>
const root = render()
expect(root.getByText('My chart')).toBeVisible()
expect(root.getByText('Loading overlay')).toBeVisible()
```

Reply:
Thanks, didn't think about it in this way :) I adjusted the test 👍